### PR TITLE
Feature 186

### PR DESCRIPTION
**Pull request by tholut**
Done
- Add models, endpoints and service for default working breaks
- Add function V2 for merge default working hours with workingTimes

### CODE REVIEW

File: src/services/default-working-breaks.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+      if (!isValid) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_FOUND,
+          `ERROR_FROM_OR_TO_NOT_IN_THE_WORKING_HOUR_RANGE`
+        );
+      }
+
+      const { ...rest } = defaultWorkingBreaksObject;
+
+      console.log(">> Create", rest);
```
Comment (akramaskar) [inline src/services/default-working-breaks.ts:112]: Remove the console.logs please before merge

File: src/services/default-working-breaks.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+      if (!isValid) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_FOUND,
+          `ERROR_FROM_OR_TO_NOT_IN_THE_WORKING_HOUR_RANGE`
+        );
+      }
+
+      const { ...rest } = update;
+
+      console.log(">> Update", rest);
```
Comment (akramaskar) [inline src/services/default-working-breaks.ts:183]: Same also here

File: src/services/default-working-hour.ts [hunk 3]
```diff
@@ -257,6 +271,63 @@ class DefaultWorkingHourService extends TransactionBaseService {
...
+
+
+  async getDefaultWorkingHourSlotTimesByLocationIdV2(locationId: string) {
+    // Todo collecting divide times dwh in 1 week with array, index 0 = sunday - index 6 = saturday
+    const dwhCollection = {};
+    const dwhList = await this.list({ location_id: locationId }, { relations: ["breaktime"], order: { "day": "ASC" }});
+    const divideBy = 5;
+
+    for (const dwh of dwhList) {
+      const dateCurr = new Date(1000 * 24 * 60 * 60 * (dwh.day + 3));
```
Comment (akramaskar) [inline src/services/default-working-hour.ts:283]: why dwh.day + 3?

File: src/services/default-working-hour.ts [hunk 4]
```diff
@@ -257,6 +271,63 @@ class DefaultWorkingHourService extends TransactionBaseService {
...
+      if (breaktime.length > 0) {
+        dwhCollection[dayIndex] = divideTimes(
+          selectSlotTimeFrom,
+          selectSlotTimeTo,
+          divideBy
+        );
+      }
+
+      const slots = [];
+      const blockedSlots = breaktime.map((bt) => ({ start: mergeMilitaryTimeWithDate(dateCurr, bt.from), end: mergeMilitaryTimeWithDate(dateCurr, bt.to)}))
```
Comment (akramaskar) [inline src/services/default-working-hour.ts:303]: Can you please add some comments here to explain your code?

**Review by akramaskar — CHANGES_REQUESTED** [review]
Need some revision please.

File: src/services/default-working-hour.ts (hunk 3 above)
Comment (tholut) [reply src/services/default-working-hour.ts:283]: It's because i want to get dateCurr from Sunday and currently unix timestamp is starting from Thursday

**Review by akramaskar — APPROVED** [review]

**Pull request merged**